waku/node/v2/quicksim2.nim

```diff
@@ -40,11 +40,11 @@ for i in 0..<topicAmount:
   os.sleep(50)
   # TODO: This would then publish on a subtopic here
   var s = "hello " & $2
-  var res3 = waitFor nodes[0].wakuPublish("waku", s & "0")
-  res3 = waitFor nodes[1].wakuPublish("waku", s & "1")
-  res3 = waitFor nodes[2].wakuPublish("waku", s & "2")
-  res3 = waitFor nodes[3].wakuPublish("waku", s & "3")
-  res3 = waitFor nodes[4].wakuPublish("waku", s & "4")
+  var res3 = waitFor nodes[0].wakuPublish("waku", cast[seq[byte]](s & "0"))
```

Comment:
Where is the protobuf? Fine to not edit examples, but should be in test at minimum

Reply:
@oskarth so I haven't tested with passing protobufs, but what we can do now is at least pass a serialized protobuf in the data field which would correspond to a status-message. Right?

Reply:
For ProtoBuf, you can use the recently developed library here:
https://github.com/status-im/nim-protobuf-serialization

You'll be able to find quite a lot of usage examples in the test folder.

Reply:
`nim-protobuf-serialization` has some fundamental semantic issues where it moves away from how protobuf is specified that are being resolved - in the meantime, it's probably safer to use minprotobuf from libp2p in waku (since waku is built on top of libp2p, staying consistent is reasonable) - when/if nps becomes compatible and libp2p moves, waku can do the same (minprotobuf is part of the security audit as well)

Reply:
@arnetheduck, you should consider that the issues in `nim-protobuf-serialization` concern only corner cases that are very much unlikely to appear in the usage patterns needed in Waku. `nim-protobuf-serialization` has a significantly simpler to use API which will make the integration here easier.

Reply:
well - when using a black box library like nps with no expectation of understanding what it does, it's important the black box does the expected thing - for example interop with other protobuf implementations correctly the way that upstream `protoc` does - libp2p, and waku by extension, uses protobuf 2 which is a more nuanced protocol, and specifically what nps has semantic issues with - at such, the API minprotobuf exposes (now) allows correctly implementing proto2 semantics - it's inherently a more simple library with less magic going on, but does indeed require a deeper understanding of protobuf itself.

Reply:
as an example, the issues already affect correct implementation of the gossipsub messages (in protobuf) - I'm not sure what you're referrring to when you say waku's usage patterns would not be affected - do you mean that when designing the waku messages, we should consider nps's limitations and use a lowest common denominator between upstream protobuf and nps-protobuf?

Reply:
>  libp2p, and waku by extension, uses protobuf 2 

I believe you are mistaken here. The gossip layer in LibP2P may use Proto2, but the Vac team is using Proto3 when defining the higher-level messages that are delivered as payloads within the gossip messages. Here is one example:

https://github.com/vacp2p/mvds/blob/master/protobuf/sync.proto

The [Nim implementation of MVDS](https://github.com/status-im/nim-mvds), which is based on `nim-protobuf-serialization`, demonstrates how easy it was to apply the new library without invoking any of the disputed extensions. If anything, we may invest a little bit of time to integrate the Proto files parser that will make the usage even more straight-forward.

Reply:
that's mvds which is part of the status protocol - waku over libp2p however will be using gossipsub or an extension thereof - gossipsub uses proto2 and uses the disputed features (dontOmit in particular) - which among other things, prevents it from being merged into libp2p

Reply:
Why would the code here then deal with ProtoBuf in any way? GossipSub is already implemented in LibP2P. I think we need a clarification from @oskarth regarding his original question.

Reply:
Well, it makes sense for the payload itself to also be protobuf, if that is what waku requires. So I see no conflict here whether using protobuf 2 or 3, or even some other serialization format.

Reply:
> `nim-protobuf-serialization` has some fundamental semantic issues where it moves away from how protobuf is specified that are being resolved - in the meantime, it's probably safer to use minprotobuf from libp2p in waku (since waku is built on top of libp2p, staying consistent is reasonable) - when/if nps becomes compatible and libp2p moves, waku can do the same (minprotobuf is part of the security audit as well)

IMO, if we want to stop parsing protobuf by hand we need to push `nim-protobuf-serialization` a little harder. I don't know what the timelines for Waku are, but if time allows we should take the opportunity to trial run it and workout whatever kinks there are in the process. Manual protobuf parsing is nuts and incredibly error prone already.

Reply:
well, we can - but it needs to implement protobuf first :/

Reply:
Good point for sure, but we got to start somewhere and that's why I'm nudging @decanus and @oskarth into it ;)

Reply:
Thanks for all the comments! This conversation is getting a bit too long for a PR subthread, let's continue it in this issue https://github.com/status-im/nim-waku/issues/58